packages-ts/starknet/src/utils.ts

```
@@ -0,0 +1,207 @@
+// It returns options for Devnet as default when nothing is configured in the environment.
+const makeFunderOptsFromEnv = (): AccountFunderOptions => {
+  const network = process.env.NETWORK || DEVNET_NAME
+  const gateway_url = process.env.NODE_URL || DEVNET_URL
+
+  return { network, gateway_url }
+}
+
+const Funder = (opts: AccountFunderOptions): AccountFunder => {
+  return new AccountFunder(opts)
+}
+
+export const account: FunderOpts = {
+  makeFunderOptsFromEnv: makeFunderOptsFromEnv,
+  Funder: Funder,
+}
+
+interface FundAccounts {
+  account: string
+  amount: number
+}
+
+interface AccountFunderOptions {
+  network?: string
+  gateway_url?: string
+}
+
+// Define the Strategy to use depending on the network.
+class AccountFunder {
+  private opts: AccountFunderOptions
+  private strategy: IFundingStrategy
+
+  constructor(opts: AccountFunderOptions) {
+    this.opts = opts
+    if (this.opts.network === DEVNET_NAME) {
+      this.strategy = new DevnetFundingStrategy()
+      return
+    }
+    this.strategy = new AllowanceFundingStrategy()
+  }
+
+  // This function adds some funds to pre-deployed account that we are using in our test.
+  public async fund(accounts: FundAccounts[]) {
+    await this.strategy.fund(accounts, this.opts)
+  }
+}
+
+interface IFundingStrategy {
+  fund(accounts: FundAccounts[], opts: AccountFunderOptions): Promise<void>
+}
+
+// Fund the Account on Devnet
+class DevnetFundingStrategy implements IFundingStrategy {
+  public async fund(accounts: FundAccounts[], opts: AccountFunderOptions) {
+    accounts.forEach(async (account) => {
+      const body = {
+        address: account.account,
+        amount: account.amount,
+        lite: true,
+      }
+
+      try {
+        await fetch(`${opts.gateway_url}/mint`, {
+          method: 'post',
+          body: JSON.stringify(body),
+          headers: { 'Content-Type': 'application/json' },
+        })
+      } catch (error: any) {}
```

How do we know if this fails? Swallowing errors like this is just a foot gun that will go off in the worst moment. If the funding strategy fails, the test should error.

I'd suggest we just remove the `try..catch` from here.

---

Yes If this fail the account will doesn't have any fund and the test will fail that's all. I remove it.